Some patches have been taken from the ace-of-penguins AUR package.
https://aur.archlinux.org/cgit/aur.git/tree/ace14_fixes.patch?h=ace-of-penguins

diff -Naur ace-1.4/config.guess ace-1.4-alpine/config.guess
--- ace-1.4/config.guess	2012-03-24 18:00:49.000000000 +0000
+++ ace-1.4-alpine/config.guess	2021-12-19 21:36:35.650013257 +0000
@@ -882,6 +882,9 @@
 	    echo ${UNAME_MACHINE}-unknown-linux-gnueabi
 	fi
 	exit ;;
+    aarch64*:Linux:*:*)
+	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit ;;
     avr32*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
@@ -950,7 +953,7 @@
 	  *)    echo hppa-unknown-linux-gnu ;;
 	esac
 	exit ;;
-    ppc64:Linux:*:*)
+    ppc64*:Linux:*:*)
 	echo powerpc64-unknown-linux-gnu
 	exit ;;
     ppc:Linux:*:*)
diff -Naur ace-1.4/lib/Makefile.am ace-1.4-alpine/lib/Makefile.am
--- ace-1.4/lib/Makefile.am	2012-03-24 18:00:49.000000000 +0000
+++ ace-1.4-alpine/lib/Makefile.am	2021-12-19 21:36:12.876679139 +0000
@@ -6,7 +6,7 @@
 CLEANFILES = images.c images.d
 
 INCLUDES = $(X_CFLAGS) @PDA@
-AM_LDFLAGS = $(X_LIBS)
+AM_LDFLAGS = $(X_LIBS) -lpng -lz -lm
 
 BUILD_CC = @BUILD_CC@
 AR = @AR@
diff -Naur ace-1.4/lib/make-imglib.c ace-1.4-alpine/lib/make-imglib.c
--- ace-1.4/lib/make-imglib.c	2012-03-24 18:00:49.000000000 +0000
+++ ace-1.4-alpine/lib/make-imglib.c	2021-12-19 21:36:12.876679139 +0000
@@ -86,7 +86,7 @@
     png_ptr = png_create_read_struct (PNG_LIBPNG_VER_STRING, 0, 0, 0);
     info_ptr = png_create_info_struct (png_ptr);
 
-    if (setjmp (png_ptr->jmpbuf)) {
+    if (setjmp (png_jmpbuf (png_ptr))) {
       fclose (f);
       continue;
     }
diff -Naur ace-1.4/lib/xwin.c ace-1.4-alpine/lib/xwin.c
--- ace-1.4/lib/xwin.c	2012-03-24 18:00:49.000000000 +0000
+++ ace-1.4-alpine/lib/xwin.c	2021-12-19 21:36:12.876679139 +0000
@@ -55,7 +55,6 @@
   { "-visual", OPTION_INTEGER, &visual_id },
   { 0, 0, 0 }
 };
-OptionDesc *xwin_options = xwin_options_list;
 
 Display *display=0;
 int screen=0;
@@ -841,13 +840,13 @@
   png_ptr = png_create_read_struct (PNG_LIBPNG_VER_STRING, 0, 0, 0);
   info_ptr = png_create_info_struct (png_ptr);
 
-  if (setjmp (png_ptr->jmpbuf)) {
+  if (setjmp (png_jmpbuf (png_ptr))) {
     fprintf(stderr, "Invalid PNG image!\n");
     return;
   }
 
   file_bytes = src->file_data;
-  png_set_read_fn (png_ptr, (voidp)&file_bytes, (png_rw_ptr)png_reader);
+  png_set_read_fn (png_ptr, &file_bytes, (png_rw_ptr)png_reader);
 
   png_read_info (png_ptr, info_ptr);
 
